Docs: Improve Open Telemetry documentation (#47955)
vercel/next.js: docs/advanced-features/open-telemetry.md

Next.js automatically instruments several spans for you to provide useful insights into your application's performance.

> Attributes on spans follow [OPenTelemetry semantic conventions](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/) and we also add a few custom attributes in `next` namespace:

Reviewer: ```suggestion
Attributes on spans follow [OpenTelemetry semantic conventions](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/). We also add some custom attributes under the `next` namespace:
```

- `next.span_type`: `AppRouteRouteHandlers.runHandler`.

This span represents the execution of an API route handler in app router.

Attributes:

- `next`
  - `next.span_name`
  - `next.span_type`
  - `next.route`

### `getServerSideProps [next.route]`

- `next.span_type`: `Render.getServerSideProps`.

This span represents the execution of `getServerSideProps` for a specific route.

Attributes:

- `next`
  - `next.span_name`
  - `next.span_type`
  - `next.route`

### `getStaticProps [next.route]`

- `next.span_type`: `Render.getStaticProps`.

This span represents the execution of `getStaticProps` for a specific route.

Attributes:

- `next`
  - `next.span_name`
  - `next.span_type`
  - `next.route`

### `render route (pages) [next.route]`

- `next.span_type`: `Render.renderDocument`.

This span represents the process of rendering the document for a specific route.

Attributes:

- `next`
  - `next.span_name`
  - `next.span_type`
  - `next.route`

### `generateMetadata [next.page]`

- `next.span_type`: `ResolveMetadata.generateMetadata`.

This span represents the process of generating metadata for a specific page (single route can have multiple of these spans).

Attributes:

> - `next`

Reviewer: I don't think we need those to be indented

Author: Removed `next` group from all spans.
